src/strings.cpp

```diff
@@ -818,15 +817,10 @@ static char *FormatString(char *buff, const char *str_arg, StringParameters *arg
 	WChar b = '\0';
 	uint next_substr_case_index = 0;
 	char *buf_start = buff;
-	std::stack<const char *, std::vector<const char *>> str_stack;
-	str_stack.push(str_arg);
-
+	const char* current_str_arg = str_arg;
 	for (;;) {
-		while (!str_stack.empty() && (b = Utf8Consume(&str_stack.top())) == '\0') {
-			str_stack.pop();
-		}
-		if (str_stack.empty()) break;
-		const char *&str = str_stack.top();
+		if ((b = Utf8Consume(&current_str_arg)) == '\0') break;
+		const char *&str = current_str_arg;
```

Comment:
This seems to be some messing about so the value in the stack gets updated properly. However, now there is only one value `current_str_arg`, so can't `const char *current_str_arg = str_arg` become `const char *str = str_arg`, remove this line and pass `str` instead of `current_str_arg` to `Utf8Consume`? That seems to simplify this complicated construct a lot.

Reply:
`str` is used 26 times in the function, thus changing it will introduce a lot of noise in this PR.
I rather change this in a separate clean-up PR.

Reply:
Keep it as a separate commit, and I'd have no issues with it being in this PR